torch/distributed/_sharding_spec/api.py

```diff
@@ -110,3 +165,8 @@ def __post_init__(self):
             rank = len(shard.shard_offsets)
 
         validate_non_overlapping_shards_metadata(self.shards)
+
+    def shard(self, tensor_sizes: List[int], process_group=None) -> List[ShardMetadata]:
+        # check if shards form a valid tensor
+        check_tensor(self.shards, tensor_sizes)
+        return self.shards
```

Comment:
N00b question, are we going to do the sharding here? Or this is actually just a no-op plus validation check?

Reply:
actually after some discussion, I think we might not able to do sharding for EnumerableShardingSpec as we don't have the clue how the shards layout look like and how to scatter the shards, just left this not implemented until we figure out a better way